### PR TITLE
Fix issue #1637: setDay with weekStartsOn != 0

### PR DESCRIPTION
**Pull request by BoomDev**
As demonstrated in the issue, setDay(date, day, [option]) returns an
incorrect value if date is on a Sunday and option.weekStartsOn is != 0.
This includes a fix (there's probably a nicer way to code this) and
2 corresponding test cases.

I also found 4 test cases that I believe to be incorrect, all using
the argument {weekStartsOn: 1}. I hope I'm not mistaken here, but I
changed the test cases accordingly.

- All tests completed

- Build script run

- Added entry to Unreleased section of CHANGELOG

### CODE REVIEW

File: src/setDay/test.js [hunk 1]
```diff
@@ -49,6 +49,20 @@ describe('setDay', function() {
...
+  it('specifies Monday as the first day of the week', function() {
+    var result = setDay(new Date(2014, 8 /* Sep */, 7), 1, {
+      weekStartsOn: '1'
+    })
+    assert.deepEqual(result, new Date(2014, 8 /* Sep */, 1))
+  })
+
+  it('specifies Tuesday as the first day of the week', function() {
+    var result = setDay(new Date(2014, 8 /* Sep */, 7), 1, {
+      weekStartsOn: '2'
```
Comment (dcousens) [inline src/setDay/test.js]: Why a string?

**Comment by dcousens** [comment]
@BoomDev please see my update to the examples in https://github.com/BoomDev/date-fns/pull/1

File: src/setDay/test.js (hunk 1 above)
Comment (BoomDev) [reply src/setDay/test.js]: Oh, good catch. In fact I copied the case just above (it 'implicitly converts options') and didn't realize it was using a string... So the answer is: no reason. I'll change it in my 2 cases.

**Comment by trollepierre** [comment]
🙏 @dcousens : can you approve this PR since the change has been taken into consideration? 
+ I opened a new bug that count on this fix

**Comment by trollepierre** [comment]
@BoomDev : have you checked if "lastDayOfWeek startOfWeek differenceInCalendarWeeks" are fixed?

**Comment by dcousens** [comment]
@trollepierre I am not a maintainer,  I am merely a fellow user wanting to help :+1:

**Comment by BoomDev** [comment]
> @BoomDev : have you checked if "lastDayOfWeek startOfWeek differenceInCalendarWeeks" are fixed?

@trollepierre I have not.

**Comment by dkozickis** [comment]
Thank you @BoomDev and @dcousens, this seems GTG. 
Tests might be failing on CI because of DST/Timezones, not sure @kossnocorp

**Comment by kossnocorp** [comment]
@dkozickis yeah, some tests were failing because of a DST issue that was fixed in master. I've rebased the branch but there's still a couple failing tests in the changed function. I'm on it.

**Pull request merged**